py/openage/convert/dataformat/exportable.py

```diff
@@ -381,7 +382,7 @@ def structs(cls):
             dbg(lazymsg=lambda: "%s: exporting member %s<%s>" % (repr(cls), member_name, member_type), lvl=3)
 
             if isinstance(member_type, MultisubtypeMember):
-                for subtype_name, subtype_class in member_type.class_lookup.items():
+                for subtype_name, subtype_class in sorted(member_type.class_lookup.items()):
```

Comment:
If you don't sort the items, their order is undefined; in practice, it varies from run to run.
The iteration order translates to the order of the segments in the generated C++ code, which will be different with each run, triggering unneeded re-builds.